vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -277,14 +334,14 @@ func matchNameforVMDK(mount_source string) bool {
 }
 
 // enumberates volumes and  builds RefCountsMap, then sync with mount info
-func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver) error {
+func (r *RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver) error {
 	// we assume to  have empty refcounts. Let's enforce
 
-	r.mtx.Lock() // Lock the RefCountsMap to purge the refcounts
-	for name := range r.refMap {
-		delete(r.refMap, name)
-	}
-	r.mtx.Unlock() // Unlock.
+	r.dirtyMtx.Lock()
+	r.isDirty = false
+	r.dirtyMtx.Unlock()
+
```

Comment:
what if Mount() sets dirty right before the lock on L340 , but the actual mount happens after this discovery thread is done talking to Docker? Won't this will reset the Dirty flag and the mount() will be unaccounted for ?